reactivesocket-cpp/src/Frame.h

```diff
@@ -122,20 +151,23 @@ class Frame_REQUEST_SUB {
       StreamId streamId,
       FrameFlags flags,
       uint32_t requestN,
+      folly::Optional<FrameMetadata> metadata,
       Payload data)
       : header_(FrameType::REQUEST_SUB, flags, streamId),
         requestN_(requestN),
+        metadata_(std::move(metadata)),
         data_(std::move(data)) {}
 
   /// For compatibility with other data-carrying frames.
-  Frame_REQUEST_SUB(StreamId streamId, FrameFlags flags, Payload data)
-      : Frame_REQUEST_SUB(streamId, flags, 0, std::move(data)) {}
+  Frame_REQUEST_SUB(StreamId streamId, FrameFlags flags, folly::Optional<FrameMetadata> metadata, Payload data)
+      : Frame_REQUEST_SUB(streamId, flags, 0, std::move(metadata), std::move(data)) {}
 
   Payload serializeOut();
   bool deserializeFrom(Payload in);
 
   FrameHeader header_;
   uint32_t requestN_;
+  folly::Optional<FrameMetadata> metadata_;
```

Comment:
It feels to me that using folly::Optional in this case is unnecessary. FrameHeader is storing a unique_ptr which can express nullability.
I would suggest removing Optional from this diff.